File: backend/internal/domain/entities/progress.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type LessonProgress struct {
	ID          uuid.UUID
	UserID      uuid.UUID
	LessonID    uuid.UUID
	OrgID       uuid.UUID
	CompletedAt *time.Time
	Score       *float64
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

File: frontend/src/types/progress.ts
export interface ProgressDTO {
  id: string
  user_id: string
  lesson_id: string
  org_id: string
  score?: number
  completed_at?: string
  created_at: string
}

export interface InsightsDTO {
  insights: string
  total_lessons: number
  completed_lessons: number
  average_score: number
}

export interface CourseProgressSummary {
  course_id: string
  total_lessons: number
  completed_lessons: number
  percentage: number
  average_score?: number
}
